code/modules/mob/living/living.dm

```diff
@@ -1591,10 +1591,11 @@ GLOBAL_LIST_EMPTY(fire_appearances)
  * Signals the extinguishing.
  */
 /mob/living/proc/extinguish_mob()
+	if(HAS_TRAIT(src, TRAIT_PERMANENTLY_ONFIRE)) //The everlasting flames will not be extinguished
+		return
```

Comment:
You should update the document comment for `TRAIT_PERMANENTLY_ONFIRE`, because you're entirely changing it's purpose. 

Prior its sole intention was that it applied the fire overlay regardless of whether the mob was actually on fire, now it entirely prevents a mob from extinguishing themself.

Reply:
Yup